jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/AbstractLayoutModel.java

```diff
@@ -75,19 +90,38 @@ public void accept(LayoutAlgorithm<N, P> layoutAlgorithm) {
       layoutAlgorithm.visit(this);
 
       if (layoutAlgorithm instanceof IterativeContext) {
-        setupVisRunner((IterativeContext) layoutAlgorithm);
-      } else if (log.isTraceEnabled()) {
-        log.trace("no visRunner for this {}", this);
+        setRelaxing(true);
+        setupVisRunner(
+            (IterativeContext) layoutAlgorithm,
+            layoutAlgorithm instanceof IterativeContext.WithPreRelax);
```

Comment:
If you decide to stick with the tagging interface, consider having two overloads of setupVisRunner(): one that takes an `IterativeContext` and one that takes an `IterativeContext.WithPreRelax`.